### PR TITLE
Api image support

### PR DESCRIPTION
**Pull request by teneightfive**
Allow images to be served from the content store. Image paths need
to follow the documentation mentioned on the content store docs:
http://nhsuk-content-store.readthedocs.io/en/latest/references.html#id3

The true url is also proxied via a local path.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -23,7 +23,7 @@
   "dependencies": {
     "JSON2": "~0.1.0",
     "applicationinsights": "~0.17.0",
-    "async": "^2.1.4",
+    "async": "~2.1.4",
```
Comment (tyom) [inline package.json]: Do you think it's safe to change it minor version update? I've seen semver misused and things getting broken because of silent update. I wonder if minor updates should be manual.

Comment (tyom) [inline package.json]: Wait, tilde is the actual minor update lock. Don't mind me.

File: app/views/_macros/components.nunjucks [hunk 2]
```diff
@@ -1,28 +1,30 @@
...
+            {{ path }} {{ width }},
+          {% endif %}
+        {% endfor %}
+      "
+      alt="{{ options.alt }}"
+      data-analytics="image"
+      data-analytics-type="{{ options.analyticsType | default('inline') }}"
+      />
+    <!--<![endif]-->
+    <!--[if gt IE 7]><!--><noscript><!--<![endif]-->
```
Comment (tyom) [inline app/views/_macros/components.nunjucks]: Seems like this IE conditional is unnecessary. Surely we can fall back to any browser?

File: app/views/_macros/components.nunjucks [hunk 3]
```diff
@@ -1,28 +1,30 @@
...
-    alt="{{ options.alt }}"
-    data-analytics="image"
-    data-analytics-type="{{ options.analyticsType | default('inline') }}"
-    />
-  <!--<![endif]-->
-  <!--[if gt IE 7]><!--><noscript><!--<![endif]-->
-    {% set fallback = options.srcset[0] | split(' ') %}
-    <img src="{{ asset_path(fallback[0]) }}" alt="{{ options.alt }}" />
-  <!--[if gt IE 7]><!--></noscript><!--<![endif]-->
+  {% if options.srcset %}
```
Comment (tyom) [inline app/views/_macros/components.nunjucks]: One of the problems with having single argument as dictionary is that it's not clear without reading the whole code block what properties should be in it. Should either document them or pass them as separate arguments with defaults (method I personally prefer).

File: app/views/_macros/components.nunjucks [hunk 4]
```diff
@@ -1,28 +1,30 @@
...
+    <!--[if gt IE 7]><!-->
+    <img
+      srcset="
+        {% for src in options.srcset %}
+          {% set splitSrc = src | split(' ') %}
+          {% set path = asset_path(splitSrc[0]) %}
+          {% set width = splitSrc[1] %}
+
+          {% if loop.last %}
```
Comment (tyom) [inline app/views/_macros/components.nunjucks]: Could do this instead:
```jinja
{% set comma = joiner() %}
{% for src in options.srcset %}
  {% set splitSrc = src | split(' ') %}
  {% set path = asset_path(splitSrc[0]) %}
  {% set width = splitSrc[1] %}

  {{ comma() }} {{ path }} {{ width }}
{% endfor %}
```

File: app/views/_macros/components.nunjucks (hunk 3 above)
Comment (teneightfive) [inline app/views/_macros/components.nunjucks]: I think the original reason for doing this was so you can't have to pass null or similar to the component if you want to send a value for the last argument but not ones before it.

Comment (teneightfive) [inline app/views/_macros/components.nunjucks]: Maybe something we could look at but I think it should be a different PR?
